### PR TITLE
Fix rasterio coordinates

### LINKED ISSUE #31: Image coordinate order
We use the coordinate order of rasterio when loading images, which is channels first (channels, rows, cols). Most image processing, visualization and machine learning libraries, however, use channels last (rows, cols, channels) for image coordinate convention (e.g. scikit-image, opencv, pillow, matplotlib, etc.).  

In order to avoid confusion and to adjust to user requirements more easily it would be great if we could provide an option to change the coordinate order upon instantiation of an image object. 

I know that Rasterio provides conversion methods [here](https://rasterio.readthedocs.io/en/latest/topics/image_processing.html?highlight=read#image-processing-software), but it would be convenient if the order could be set once (e.g. as env var or parameter in Image init) to comply with a user's requirements.

Tensorflow for example provides such a switch between channels first and channels last to account for performance differences when training on GPU or CPU.

### PR DESCRIPTION
**Pull request by MWieland**
closes #31

### CODE REVIEW

File: ukis_pysat/raster.py [hunk 1]
```diff
@@ -27,32 +28,43 @@
 
 
 class Image:
-    def __init__(self, path=None, dataset=None, arr=None):
+    def __init__(self, path=None, dataset=None, arr=None, dimorder='first'):
```
Comment (fwfichtner) [inline ukis_pysat/raster.py:31]:
```suggestion
    def __init__(self, path=None, dataset=None, arr=None, dimorder="first"):

```

File: ukis_pysat/raster.py [hunk 2]
```diff
@@ -27,32 +28,43 @@
...
                 raise TypeError(
                     f"dataset must be of type rasterio.io.DatasetReader and arr must be of type " f"numpy.ndarray"
                 )
+        self.dimorder = dimorder
         self.transform = self.dataset.transform
         self.crs = self.dataset.crs
         self.da_arr = None
 
+    @property
+    def arr(self):
```
Comment (fwfichtner) [inline ukis_pysat/raster.py:52]: I would prefer it like this:
```python
    def arr(self):
        if self.dimorder == "first":
            return self.__arr
        elif self.dimorder == "last":
            return reshape_as_image(self.__arr)
        else:
            raise AttributeError("dimorder for bands or channels must be either 'first' or 'last'.")
```

File: ukis_pysat/raster.py (hunk 1 above)
Comment (fwfichtner) [reply ukis_pysat/raster.py:31]: black

File: tests/test_raster.py [hunk 3]
```diff
@@ -30,6 +31,41 @@ def test_init(self):
...
+                    11.9027457562112939,
+                    51.4664152338322580,
+                    11.9477435281016131,
+                    51.5009522690838750,
+                    img_last.arr.shape[1],
+                    img_last.arr.shape[0],
+                )
+            ),
+        )
+
```
Comment (fwfichtner) [inline tests/test_raster.py:68]: We should test the error as well.

**Comment by MWieland** [comment]
Ok, will update accordingly.

**Pull request merged**